ToolChain/Assembler/Parser/Parser.py

```diff
@@ -446,20 +346,80 @@ def _evaluateAndExtractInfoFromLine(self, line, buildInstruction):
                     part = part[1:]
                 if not foundSource:
                     foundSource = True
+
                     # Verify this reference doesn't have additonal ':', hotfix for issue 18
                     if part.count(':') > 0:
                         raise ValueError("Syntax error, memory reference has too many {}".format(MEMORY_REFERENCE_INDICATORS))
-                    buildInstruction.sourceImmediate = MEMORY_REFERENCE_INDICATORS+ part.upper() + MEMORY_REFERENCE_INDICATORS
+                    buildInstruction.sourceImmediate = MEMORY_REFERENCE_INDICATORS + part.upper() + \
+                                                       MEMORY_REFERENCE_INDICATORS
```

Comment:
In this case, for readability, I would put part.upper() on a different line as well.